benchto-driver/src/main/java/io/trino/benchto/driver/execution/BenchmarkExecutionDriver.java

```diff
@@ -196,12 +275,17 @@ private QueryExecutionResult executeSingleQuery(
         }
 
         macroService.runBenchmarkMacros(benchmark.getAfterExecutionMacros(), benchmark, connection);
-        return result;
+        return Optional.of(result);
     }
 
     private Connection getConnectionFor(QueryExecution queryExecution)
             throws SQLException
     {
         return applicationContext.getBean(queryExecution.getBenchmark().getDataSource(), DataSource.class).getConnection();
     }
+
+    private boolean isTimeLimitExceeded(Optional<ZonedDateTime> executionTimeLimit)
```

Comment:
nit: time linit change should be a separate commit

Reply:
Currently time limit is only used in concurrency benchmark (for standard we pass empty Optional as time limit). So this would need to be a strange no-op change to introduce it separately.